contracts/smart-account/modules/SessionValidationModules/ERC20SessionValidationModule.sol

```diff
@@ -79,4 +79,35 @@ contract ERC20SessionValidationModule is ISessionValidationModule {
                 _sessionKeySignature
             ) == sessionKey;
     }
+
+    /**
+     * @dev validates that the call (destinationContract, callValue, funcCallData)
+     * complies with the Session Key permissions represented by sessionKeyData
+     * @param destinationContract address of the contract to be called
+     * @param callValue value to be sent with the call
+     * @param _funcCallData the data for the call. is parsed inside the SVM
+     * @param _sessionKeyData SessionKey data, that describes sessionKey permissions
+     */
+    function validateSessionParams(
```

Comment:
can we make internal method for common code which can be used by this method and above validateSessionUserOp method

Reply:
The way we parse calldata is different, so the only code we can move to an internal method would be the checks.
Adding another method could add some extra gas I think, so I suggest leaving it as is, as the module is relatively small and it's not complicated to track changes.
Also it would require passing 8 parameters to the internal method

Reply:
fair enough